### PR TITLE
Change API and framework Active Response message format to JSON

### LINKED ISSUE #7210: Change Active Response messages to new JSON format
|Wazuh version|Component|
|---|---|
| 4.2.0 | API/Framework |

As explained in https://github.com/wazuh/wazuh/issues/6954, Active Response messages need to be changed to JSON format to remove the limitation of the alert fields they can receive.

Some API endpoints will be affected:

- `/cluster/configuration/validation` -> `check-manager-configuration` AR message sent to `execd`
- `/manager/configuration/validation` -> `check-manager-configuration` AR message sent to `execd`

```json
{
  "version": 1,
  "origin": {
    "module": "api"
  },
  "command": "check-manager-configuration",
  "parameters": {
    "extra_args": [],
    "alert": {}
  }
}
```

- `/cluster/restart` -> `restart-wazuh` AR message sent to `execd`
- `/manager/restart` -> `restart-wazuh` AR message sent to `execd`

```json
{
  "version": 1,
  "origin": {
    "module": "api"
  },
  "command": "restart-wazuh",
  "parameters": {
    "extra_args": [],
    "alert": {}
  }
}
```

- `/agents/restart` -> AR message sent to `arq`
- `/agents/{agent_id}/restart` -> AR message sent to `arq`
- `/agents/group/{group_id}/restart` -> AR message sent to `arq`

```json
{
  "version": 1,
  "origin": {
    "module": "api"
  },
  "command": "restart-wazuh0",
  "parameters": {
    "extra_args": [],
    "alert": {}
  }
}
```

- `/active-response` -> AR message sent to `arq`

Example 1:

```json
{
  "version": 1,
  "origin": {
    "module": "api"
  },
  "command": "firewall-drop",
  "parameters": {
    "extra_args": [],
    "alert": {
      "data": {
        "srcip": "1.1.1.1"
      }
    }
  }
}
```

Example 2:

```json
{
  "version": 1,
  "origin": {
    "module": "api"
  },
  "command": "disable-account",
  "parameters": {
    "extra_args": [],
    "alert": {
      "data": {
        "dstuser": "user1"
      }
    }
  }
}
```

Example 3:

```json
{
  "version": 1,
  "origin": {
    "module": "api"
  },
  "command": "custom-ar",
  "parameters": {
    "extra_args": ["--argv1","--argv2"],
    "alert": {
      "data": {
        "custom_key1": "custom_value1",
        "custom_key2": "custom_value2"
      }
    }
  }
}
```

### Notes:

- Restart agents: `restart-ossec0` AR has been renamed to `restart-wazuh0`.
- JSON message: `version`, `origin`, `command`, `parameters`, `parameters.extra_args` (array) and `parameters.alert` (object) are mandatory, `parameters.alert.data` (object) is optional and its information depends on the AR executed.
- Messages sent to `arq` will change only the last part. Flags and `agent_id` information will not change: `(msg_to_agent) [] NNS 001 {JSON_message}`.
- Backward compatibility: framework needs to identify the agent version where the AR will be placed. If `agent_version < 4.2.0`, message should be in string format (with no changes). If `agent_version >= 4.2.0`, new message should be sent.
- Messages sent to all agents (no `agent_id` specified) need to be sent individually to each agent with the format depending on its version (as explained in the last point): `(msg_to_agent) [] NNS 001 'string_message'|{JSON_message}`.
- These changes only affect AR messages. No AR messages (for example, `HC_SK_RESTART`) will remain the same: `(msg_to_agent) [] N!S 001 syscheck restart`.
- Since AR are now executables, they have changed their names:
  - Linux example: `firewall-drop.sh` -> `firewall-drop`
  - Windows example: `netsh.cmd` -> `netsh.exe`

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by mcarmona99**
|Related issue|
|---|
| #7210 |

This pull request closes #7210.

In this PR, I have changed the Active Response messages to JSON when needed.

The message sent in `/cluster/configuration/validation`, `/manager/configuration/validation`, `/cluster/restart` and `/manager/restart` has been changed to the standard JSON format, as detailed in the issue.

The message sent in `/agents/restart`, /`agents/{agent_id}/restart` and `/agents/group/{group_id}/restart` changes to JSON format when the agent version is `>= 4.2.0`.

The message sent in `/active-response` will be the same if the agent version is `< 4.2.0` and it will be a JSON message when agent version is `>=4.2.0`. The message is built using the JSON body when using the API. I have included a new field in the JSON body: `alert`.

I have updated the active-response integration tests (rbac and no rbac) with the new exception added and I have sent the command `restart-wazuh0` instead of `restart-ossec0`. This change has been also added to the `ar` configurations (`ar.conf` and `mergedx.mg`).

The following unit tests have been updated too:

	pytest wazuh/core/tests/test_agent.py -> added versions and test if the message sent is the correct one
		======================================== 153 passed, 11 warnings in 1.05s =========================================

	pytest wazuh/core/tests/test_configuration.py -> added ar command restart-wazuh0 for scripts
		=============================================== 34 passed in 0.16s ================================================

	pytest wazuh/core/tests/test_active_response.py -> added test case for create_json_message function
		=============================================== 17 passed in 0.09s ================================================

	pytest  framework/wazuh/tests/test_active_response.py -> update test for run_command
		=============================================== 12 passed in 0.10s ===============================================

### Test results:

```
test_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_black_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_white_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings
```

```
test_cluster_endpoints.tavern.yaml 
	 52 passed, 1 xpassed, 70 warnings
	 
test_manager_endpoints.tavern.yaml 
	 33 passed, 36 warnings
	 
test_agent_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings
```

```
pytest wazuh/core/cluster/tests/
================================ 55 passed, 2 warnings in 0.40s =================================

pytest wazuh/core/cluster/dapi/tests/
================================ 22 passed, 11 warnings in 0.54s ================================
```

Regards,
Manuel.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,11 @@ All notable changes to this project will be documented in this file.
 
 - **API:**
   - Added new endpoint to get agent stats from different components. ([#7128](https://github.com/wazuh/wazuh/issues/7128))
+  - Change Active Response endpoint function to create messages with new JSON format. ([#7312](https://github.com/wazuh/wazuh/pull/7312))
```
Comment (davidjiglesias) [inline CHANGELOG.md:14]: Use past tense. Move to appropriate tag (Changed, not Added). Both entries (API and framework) are too similar, if there is no justification for the framework entry changes, remove it.

File: api/api/models/active_response_model.py [hunk 2]
```diff
@@ -10,75 +10,113 @@
...
-        :param arguments: Command arguments
-        :type arguments: str
+        Parameters
+        ----------
+        command : str
+            Command running in the agent. If this value starts by !, then it refers to a script name instead of a command name
+        custom : bool
+            Whether the specified command is a custom command or not
+        arguments : List[str]
+            Command arguments
```
Comment (davidjiglesias) [inline api/api/models/active_response_model.py:23]: Missing alert

File: framework/wazuh/core/active_response.py [hunk 3]
```diff
@@ -61,10 +107,16 @@ def shell_escape(command):
...
-    :param msg_queue: Message previously created, contains what is necessary to launch the active response command
-    in the agent.
-    :param agent_id: Run AR command in the agent.
-    :return: WazuhResult.
+    Parameters
+    ----------
+    msg_queue : str
+        Message previously created, contains what is necessary to launch the active response command in the agent
+    agent_id : str
+        ID specifying the agent the msg_queue will be sent to
```
Comment (davidjiglesias) [inline framework/wazuh/core/active_response.py:115]:
```suggestion
        ID specifying the agent where the msg_queue will be sent to
```

File: framework/wazuh/core/manager.py [hunk 4]
```diff
@@ -273,8 +273,8 @@ def validate_ossec_conf():
         api_socket_path = join(common.ossec_path, api_socket_relative_path)
         execq_socket_path = common.EXECQ
         # Message for checking Wazuh configuration
-        execq_msg = 'check-manager-configuration '
-
+        execq_msg = "{\"version\": 1, \"origin\": {\"module\": \"api\"}, \"command\": " \
```
Comment (davidjiglesias) [inline framework/wazuh/core/manager.py:276]: Same as above

File: framework/wazuh/core/tests/test_active_response.py [hunk 5]
```diff
@@ -28,6 +27,14 @@ def agent_info(expected_exception):
         return {'status': 'active'}
 
 
+def agent_info_exception_and_version(expected_exception, version):
+    """Returns dict to cause or not a exception code 1651 on active_response.send_command()."""
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_active_response.py:31]:
```suggestion
    """Return dict to cause or not a exception code 1651 on active_response.send_command()."""
```

File: framework/wazuh/core/tests/test_active_response.py [hunk 6]
```diff
@@ -76,6 +83,44 @@ def test_create_message(expected_exception, command, arguments, custom):
...
+@pytest.mark.parametrize('expected_exception, command, arguments, alert', [
+    (1650, None, [], None),
+    (None, 'restart-wazuh0', [], None),
+    (None, 'restart-wazuh0', [], None),
+    (None, 'restart-wazuh0', ["arg1", "arg2"], None),
+    (None, 'custom-ar', ["arg1", "arg2"], {"data": {"srcip": "1.1.1.1"}})
+])
+@patch('wazuh.core.common.ossec_path', new=test_data_path)
+def test_create_json_message(expected_exception, command, arguments, alert):
+    """Check if the json message returned is correct
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_active_response.py:95]:
```suggestion
    """Check if the returned json message is correct.
```

File: framework/wazuh/core/tests/test_active_response.py [hunk 7]
```diff
@@ -76,6 +83,44 @@ def test_create_message(expected_exception, command, arguments, custom):
...
+
+    Checks if json message returned by create_json_message(...) contains the
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_active_response.py:97]:
```suggestion
    Checks if the json message returned by create_json_message(...) contains the
```

File: framework/wazuh/core/tests/test_agent.py [hunk 8]
```diff
@@ -1461,11 +1466,26 @@ def test_calculate_status(last_keep_alive, pending, expected_status):
...
+@pytest.mark.parametrize('agents_list, versions_list', [
+    (['001', '002', '003', '004'], [{'version': ver} for ver in ['v4.2.0', 'v4.0.0', 'v4.2.1', 'v3.13.2']])
+])
 @patch('wazuh.core.agent.OssecQueue')
-def test_send_restart_command(mock_ossec_queue):
-    """Test that restart_command calls send_msg_to_agent with correct params"""
-    send_restart_command('001')
-    mock_ossec_queue.return_value.send_msg_to_agent.assert_called_once_with(ANY, '001')
+def test_send_restart_command(mock_ossec_queue, agents_list, versions_list):
+    """Test that restart_command calls send_msg_to_agent with correct params
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_agent.py:1474]:
```suggestion
    """Test that restart_command calls send_msg_to_agent with correct params.
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mcarmona99) [reply CHANGELOG.md:14]: corrected, sentence moved to `Changed` and removed the framework change entry

File: api/api/models/active_response_model.py (hunk 2 above)
Comment (mcarmona99) [reply api/api/models/active_response_model.py:23]: corrected

File: api/test/integration/test_active_response_endpoints.tavern.yaml [hunk 9]
```diff
@@ -6,7 +6,7 @@ marks:
 
 stages:
 
-  - name: Runs an Active Response command on a specified agent
+  - name: Runs an Active Response command on a specified 4.2.0 agent
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply api/test/integration/test_active_response_endpoints.tavern.yaml:9]: corrected

File: framework/wazuh/active_response.py [hunk 10]
```diff
@@ -3,36 +3,55 @@
...
+    agent_list : list
+        Agents list that will run the AR command.
+    command : str
+        Command running in the agents. If this value starts by !, then it refers to a script name instead of a command name
+    custom : bool
+        Whether the specified command is a custom command or not.
+    arguments : list
+        Command arguments.
+    alert : dict
+        Same alert from alerts.json file in JSON format.
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/active_response.py:29]: improved

File: framework/wazuh/core/cluster/utils.py [hunk 11]
```diff
@@ -153,8 +153,9 @@ def manager_restart():
     try:
         # execq socket path
         socket_path = common.EXECQ
-        # msg for restarting Wazuh manager
-        msg = 'restart-wazuh '
+        # json msg for restarting Wazuh manager
+        msg = "{\"version\": 1, \"origin\": {\"module\": \"api/framework\"}, \"command\": \"restart-wazuh\"," \
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/cluster/utils.py:157]: used json.dumps

File: framework/wazuh/core/manager.py (hunk 4 above)
Comment (mcarmona99) [reply framework/wazuh/core/manager.py:276]: used json.dumps

File: api/test/integration/test_active_response_endpoints.tavern.yaml [hunk 12]
```diff
@@ -208,28 +212,42 @@ stages:
...
           failed_items:
             - error:
                 code: 1651
               id:
                 - '009'
                 - '010'
+            - error:
+                code: 1652
+              id:
+                - '000'
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply api/test/integration/test_active_response_endpoints.tavern.yaml:239]: ar module refactored

File: api/test/integration/test_active_response_endpoints.tavern.yaml [hunk 13]
```diff
@@ -208,28 +212,42 @@ stages:
...
+              id:
                 - '011'
                 - '012'
             - error:
                 code: 1750
               id:
                 - '003'
-          total_affected_items: 8
-          total_failed_items: 5
+          total_affected_items: 7
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply api/test/integration/test_active_response_endpoints.tavern.yaml:249]: ar module refactored

File: api/test/integration/test_rbac_black_active_response_endpoints.tavern.yaml [hunk 14]
```diff
@@ -144,19 +144,26 @@ stages:
...
-        error: !anyint
+        error: 2
         data:
           affected_items: !anything
           failed_items:
             - error:
                 code: 1651
               id:
                 - '010'
+            - error:
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply api/test/integration/test_rbac_black_active_response_endpoints.tavern.yaml:160]: ar module refactored

File: api/test/integration/test_rbac_white_active_response_endpoints.tavern.yaml [hunk 15]
```diff
@@ -146,19 +146,22 @@ stages:
...
-        error: !anyint
+        error: 2
         data:
           affected_items: !anything
           failed_items:
             - error:
                 code: 1651
               id:
                 - '009'
+            - error:
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply api/test/integration/test_rbac_white_active_response_endpoints.tavern.yaml:162]: ar module refactored

File: framework/wazuh/active_response.py [hunk 16]
```diff
@@ -3,36 +3,55 @@
...
+                msg_queue = active_response.create_json_message(command=command, arguments=arguments, alert=alert)
+            else:
+                msg_queue = active_response.create_message(command=command, arguments=arguments, custom=custom)
             active_response.send_command(msg_queue, oq, agent_id)
             result.affected_items.append(agent_id)
             result.total_affected_items += 1
         except WazuhException as e:
             result.add_failed_item(id_=agent_id, error=e)
+        except KeyError:
+            result.add_failed_item(id_=agent_id, error=WazuhError(1657))
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/active_response.py:54]: ar module refactored

File: framework/wazuh/core/agent.py [hunk 17]
```diff
@@ -1131,7 +1131,13 @@ def send_restart_command(agent_id):
...
     """
     oq = OssecQueue(common.ARQUEUE)
-    ret_msg = oq.send_msg_to_agent(OssecQueue.RESTART_AGENTS, agent_id)
+
+    # If the Wazuh agent version is >= 4.2.0, the message sent will have JSON format
+    agent_version = Agent(agent_id).get_basic_information(select=['version'])['version']
+    if WazuhVersion(agent_version) >= WazuhVersion('Wazuh v4.2.0'):
+        ret_msg = oq.send_msg_to_agent(OssecQueue.RESTART_AGENTS_JSON, agent_id)
+    else:
+        ret_msg = oq.send_msg_to_agent(OssecQueue.RESTART_AGENTS, agent_id)
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/agent.py:1140]: ar module refactored

File: framework/wazuh/core/exception.py [hunk 18]
```diff
@@ -211,6 +211,8 @@ class WazuhException(Exception):
                'remediation': 'Please, visit the official documentation '
                               f'(https://documentation.wazuh.com/{WAZUH_VERSION}/user-manual/api/reference.html#tag/active-response) '
                               'to get more information about `active-response` API call'},
+        1657: {'message': 'Active response - Cannot send AR command, agent version is null',
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/exception.py:214]: ar module refactored, exception deleted

File: framework/wazuh/core/ossec_queue.py [hunk 19]
```diff
@@ -15,6 +15,8 @@ class OssecQueue:
     # Messages
     HC_SK_RESTART = "syscheck restart"  # syscheck restart
     RESTART_AGENTS = "restart-ossec0"  # Agents, not manager (000)
+    RESTART_AGENTS_JSON = "{\"version\": 1,\"origin\": {\"module\": \"api/framework\"}, \"command\": \"restart-wazuh0\", " \
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/ossec_queue.py:18]: changed to json.dumps, import json too

**Comment by mcarmona99** [comment]
Refactor done to the active response module.

I have change a lot of unit tests and api integration tests due to the fact that a lot of core and SDK functions have changed.

**Test results:**

framework/wazuh/tests/test_active_response.py
========================= 12 passed in 0.10s =========================

framework/wazuh/core/tests/test_configuration.py
========================= 34 passed in 0.15s =========================

framework/wazuh/core/tests/test_agent.py
================== 153 passed, 11 warnings in 1.04s ==================

framework/wazuh/core/tests/test_active_response.py
========================= 14 passed in 0.08s =========================

Regards,
Manuel.

**Comment by mcarmona99** [comment]
Big changes to the AR response module.

I have used the `create_wazuh_socket_message` function to create AR messages sent to `execq` and `arq`.

I have updated the docstrings with the new standard.

Function parameters now include their type and default value. Functions are also including the returned type.

I have also changed hardcoded checks, variables `AR_LEGACY_VERSION`, `RESTART_WAZUH_COMMAND`, etc have been added.



### **Test results:**

```
framework/wazuh/core/tests/test_active_response.py
============================== 14 passed in 0.08s ==============================

framework/wazuh/tests/test_active_response.py
============================== 12 passed in 0.11s ==============================

framework/wazuh/core/tests/test_configuration.py
============================== 34 passed in 0.14s ==============================                        

framework/wazuh/core/tests/test_agent.py 
======================= 153 passed, 11 warnings in 0.93s =======================
```

**Comment by mcarmona99** [comment]
### **API integration tests results:**

```
test_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_black_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings

test_rbac_white_active_response_endpoints.tavern.yaml 
	 2 passed, 4 warnings
```

File: framework/wazuh/core/tests/test_agent.py [hunk 20]
```diff
@@ -561,21 +564,21 @@ def test_agent_get_key_ko(socket_mock, send_mock):
...
-    """Tests if method restart calls other methods with correct params"""
+    """Test if method restart calls other methods with correct params."""
     with patch('wazuh.core.agent.Agent.getconfig', return_value={'active-response': {'disabled': 'no'}}) as \
             mock_config:
         agent = Agent(0)
         agent.restart()
 
         # Assert methods are called with correct params
-        mock_config.assert_called_once_with('com', 'active-response')
+        mock_config.assert_called_once_with('com', 'active-response', 'Wazuh v3.9.0')
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_agent.py:574]: I see Wazuh vX.Y.Z in some places and just vX.Y.Z, please unify, i would just use vX.Y.Z

File: framework/wazuh/core/tests/test_agent.py [hunk 21]
```diff
@@ -1408,15 +1413,21 @@ def test_agent_getconfig(socket_mock, send_mock, mock_ossec_socket):
...
+    with pytest.raises(WazuhError, match=".* 1101 .*"):
+        agent.getconfig('invalid_component', 'active-response', 'v4.0.0')
+
+    # Component or config is none
+    agent = Agent('003')
+    with pytest.raises(WazuhError, match=".* 1307 .*"):
+        agent.getconfig('com', None, 'v4.0.0')
+        agent.getconfig(None, 'active-response', 'v4.0.0')
 
     # Agent Wazuh version is lower than v3.7.0
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_agent.py:1427]: Refer to the variable instead of v3.7.0

File: framework/wazuh/core/tests/test_agent.py [hunk 22]
```diff
@@ -1461,11 +1472,26 @@ def test_calculate_status(last_keep_alive, pending, expected_status):
...
+    agents_list : list
+        List of agents' ids to test the send restart command with.
+    versions_list : list
+        List of agents' versions to test whether the message sent was the correct one or not.
+    """
+    with patch('wazuh.core.agent.Agent.get_basic_information', side_effect=versions_list):
+        for agent_id, agent_version in zip(agents_list, versions_list):
+            send_restart_command(agent_id, agent_version['version'])
+            expected_msg = mock_ossec_queue.RESTART_AGENTS_JSON if WazuhVersion(
+                agent_version['version']) >= WazuhVersion('Wazuh v4.2.0') else mock_ossec_queue.RESTART_AGENTS
```
Comment (davidjiglesias) [inline framework/wazuh/core/tests/test_agent.py:1493]: Use variable for version

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: framework/wazuh/active_response.py [hunk 23]
```diff
@@ -3,36 +3,62 @@
...
+        Whether the specified command is a custom command or not.
+    arguments : list
+        Command arguments.
+    alert : dict
+        Alert information depending on the AR executed.
 
-@expose_resources(actions=['active-response:command'], resources=['agent:id:{agent_list}'])
-def run_command(agent_list=None, command=None, arguments=None, custom=False):
-    """Run AR command in a specific agent
+    Raises
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/active_response.py:33]: Removed raises from docstring

File: framework/wazuh/core/active_response.py [hunk 24]
```diff
@@ -30,46 +49,124 @@ def create_message(command, custom, arguments):
...
+    # Once we know the agent is active, store version
+    agent_version = agent_info['version']
+
+    # Check if AR is enabled
+    agent_conf = Agent(agent_id).getconfig('com', 'active-response', agent_version)
+    if agent_conf['active-response']['disabled'] == 'yes':
+        raise WazuhError(1750)
+
+    # Create classic msg or JSON msg depending on the agent version
+    if WazuhVersion(agent_version) >= WazuhVersion('Wazuh v4.2.0'):
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/active_response.py:128]: The variable was created, I didn't change the version to the variable there.

File: framework/wazuh/core/agent.py [hunk 25]
```diff
@@ -1059,18 +1069,31 @@ def unset_single_group_agent(agent_id, group_id, force=False):
...
+        ------
+        WazuhError(1735)
+            The agent version is older than the minimum required version.
+
+        Returns
+        -------
+        dict
+            Agent's active configuration.
+        """
+        required_version = 'v3.7.0'
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/agent.py:1094]: Added to common

File: framework/wazuh/core/tests/test_agent.py (hunk 20 above)
Comment (mcarmona99) [reply framework/wazuh/core/tests/test_agent.py:574]: Unification: I have used Wazuh vX.Y.Z in all places.

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**